pkg/neighborhood/miner.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package neighborhood
```

Comment:
I'd suggest moving this `neighborhood` package to the `util` package, as a subpackage.